ConwaysGameOfLife/Generator.cs

```diff
@@ -36,35 +36,37 @@ public Universe GenerateNewUniverse()
         {
             var oldCells = _oldUniverse.Cells;
             var newCells = new List<Cell>();
-            foreach (var cell in oldCells)
-            {
-                GenerateCells(newCells, cell);
-            }
+            GenerateCells(oldCells, newCells);
             return new Universe(_oldUniverse.GridWidth, _oldUniverse.GridLength, newCells);
         }
 
-        private void GenerateCells(List<Cell> cells, Cell cell)
+        private List<Cell> GenerateCells(List<Cell> oldCells, List<Cell> newCells)
         {
-            var newCell = GenerateNewCell(cell);
-            cells.Add(newCell);
+            foreach (var cell in oldCells)
+            {
+                Cell newCell = GenerateNewCell(cell);
+                newCells.Add(newCell);
+            }
+            return newCells;
         }
 
         private Cell GenerateNewCell(Cell cell)
         {
             var numberOfLiveNeighbours = _oldUniverse.CountLiveNeighbours(cell);
             var state = cell.CellState;
+
             var newCell = new Cell(state, cell.Location.X, cell.Location.Y);
-            ShouldSwitchCellState(numberOfLiveNeighbours, state, newCell);
+            if (ShouldSwitchCellState(numberOfLiveNeighbours, state, newCell))

```

Comment:
```suggestion
            if (ShouldSwitchCellState(numberOfLiveNeighbours, state))
```

no need to pass in the `newCell` anymore as we've moved the side-effect of switching the `newCell`'s state to a separate, explicit call if the boolean is true